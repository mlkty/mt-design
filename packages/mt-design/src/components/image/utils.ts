
export const isValidImage = (src: string) => {
    return new Promise(resolve => {
        const image = new Image();
        image.onload = () => resolve(true);
        image.onerror = () => resolve(false);
        image.src = src;
    });
};
