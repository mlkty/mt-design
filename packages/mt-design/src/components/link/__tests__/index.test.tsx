import {Link} from '..';
import {customRender} from '../../../../tests/utils';
import mountTest from '../../../../tests/mountTest';

describe('Link', () => {
    mountTest(Link);

    it('render empty link', async () => {
        const {container} = customRender(<Link />);
        expect(container).toMatchSnapshot();
    });

    it('render children', async () => {
        const {container} = customRender(<Link>link</Link>);
        expect(container).toHaveTextContent('link');
    });

    const types = ['default', 'primary'] as const;
    types.forEach(type => {
        it(`render type: ${type}`, async () => {
            const {container, rerender} = customRender(<Link type={type}>link</Link>);
            expect(container.firstChild).toHaveClass(`mt-link--${type}`);
            rerender(<Link type={type} disabled>link</Link>);

            expect(container.firstChild).toHaveClass('mt-link--disabled');
        });
    });
});
