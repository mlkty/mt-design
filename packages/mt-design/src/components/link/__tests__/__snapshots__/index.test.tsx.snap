// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link render empty link 1`] = `
<div>
  <a
    class="mt-link mt-link--default mt-link--underline-hover"
  >
    <span
      class="mt-link-inner"
    />
  </a>
</div>
`;
